app/controllers/application_controller.rb

```diff
@@ -7,7 +7,11 @@ class ApplicationController < ActionController::Base
   helper_method :current_organisation, :super_admin?
 
   def current_organisation
-    @current_organisation ||= current_user.organisation
+    if session[:organisation_id] && session[:organisation_id] == current_user.organisation.id
+      Organisation.find(session[:organisation_id])
```

Comment:
This will be of more use when users have a many to many relationship with organisations

Reply:
Have we intentionally lost the `@current_organisation` cacheing?

Reply:
Yes this memoization breaks the new implementation.
Old implementation was fine because it never changed.
Have a note to come back to this.